### PR TITLE
[#1494] Popular services fix

### LINKED ISSUE #1494: Fix popular services
**Describe the bug**
Now, the popular services section contains randomly selected services. We need to fix this bug. 

**To Reproduce**
1. Go to https://marketplace.eosc-portal.eu/
2. Scroll down to 'Popular services'
4. See services

**Expected behavior**
We should display services in the "Popular services" section based on popularity: service views/service orders.

### PR DESCRIPTION
**Pull request by goreck888**
Refactoring of GA client
Add popularity_ratio to the service updated once a day

Fixes #1494

### CODE REVIEW

File: app/models/google/analytics.rb [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+      Google::Auth::ServiceAccountCredentials.make_creds(json_key_io: key,
+                                                         scope: "https://www.googleapis.com/auth/analytics.readonly")
+    rescue StandardError
+      Rails.logger.warn("[WARN] Cannot make credentials for GA")
+    end
+
+    def key
+      path = ENV["GOOGLE_AUTH_KEY_FILEPATH"] || "config/google_api_key.json"
+      File.open(path)
+    rescue StandardError
```
Comment (martaswiatkowska) [inline app/models/google/analytics.rb:37]: Please add => e

File: app/models/google/analytics.rb [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  rescue StandardError => e
+    print e
+    Rails.logger.warn("[WARN] Cannot connect to GA API")
+  end
+
+  private
+    def auth
+      Google::Auth::ServiceAccountCredentials.make_creds(json_key_io: key,
+                                                         scope: "https://www.googleapis.com/auth/analytics.readonly")
+    rescue StandardError
```
Comment (martaswiatkowska) [inline app/models/google/analytics.rb:30]: Please add => e

File: db/schema.rb [hunk 3]
```diff
@@ -404,6 +405,26 @@
     t.datetime "updated_at", null: false
   end
 
+  create_table "user_categories", force: :cascade do |t|
```
Comment (martaswiatkowska) [inline db/schema.rb:408]: Is this a part of this pr?

File: lib/tasks/update_popularity_ratio.rake [hunk 4]
```diff
@@ -0,0 +1,18 @@
+# frozen_string_literal: true
+
+namespace :services do
+  task update_popularity_ratio: :environment do
+    puts "Service popularity ratio update"
+    analytics = Google::Analytics.new
+    total_views = Analytics::TotalServicesViews.new(analytics).call.to_d
+    total_project_items = ProjectItem.all.size > 0 ? ProjectItem.all.size : 1
+    Service.find_each do |service|
```
Comment (martaswiatkowska) [inline lib/tasks/update_popularity_ratio.rake:9]: why find each not each?

Comment (goreck888) [reply lib/tasks/update_popularity_ratio.rake:9]: Service.each won't work.

File: db/schema.rb (hunk 3 above)
Comment (goreck888) [reply db/schema.rb:408]: Schema changes removed

File: app/models/google/analytics.rb (hunk 2 above)
Comment (goreck888) [reply app/models/google/analytics.rb:30]: Added

File: app/models/google/analytics.rb (hunk 1 above)
Comment (goreck888) [reply app/models/google/analytics.rb:37]: Added

**Pull request merged**